src/hotspot/os/linux/os_linux.cpp

```diff
@@ -3960,6 +3972,11 @@ char* os::Linux::reserve_memory_special_huge_tlbfs(size_t bytes,
                                                    size_t page_size,
                                                    char* req_addr,
                                                    bool exec) {
+  assert(page_size > (size_t)os::vm_page_size(), "large page size: "
+                                                        SIZE_FORMAT " not larger than small page size: "
+                                                        SIZE_FORMAT,
+                                                        page_size,
+                                                        (size_t)os::vm_page_size());
```

Comment:
I would prefer keeping the assert message short as in the other cases.
```suggestion
  assert(page_size > (size_t)os::vm_page_size(), "Must be a large page size");
```
Please also move it down to below the other assert checking `_page_sizes.contains(...)`